### PR TITLE
Refactor eventstore adapters and eliminate technical debts

### PR DESCRIPTION
**Pull request by TSPMpix**

### CODE REVIEW

File: packages/adapters/eventstore-adapters/resolve-eventstore-lite/test/__snapshots__/drop.unit.test.ts.snap [hunk 1]
```diff
@@ -3,7 +3,34 @@
 exports[`secrets store dropped: drop table with keys 1`] = `
 Array [
   Array [
-    "DROP TABLE IF EXISTS \\"secrets-table-escaped\\"",
+    Object {
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-lite/test/__snapshots__/drop.unit.test.ts.snap:6]: Bad test

File: packages/adapters/eventstore-adapters/resolve-eventstore-lite/src/init.ts [hunk 2]
```diff
@@ -1,34 +1,90 @@
...
+    )
+    log.debug(`events database tables are initialized`)
+
+    await database.exec(
+      `CREATE TABLE ${escapeId(snapshotsTableName)} (
+        ${escapeId('snapshotKey')} TEXT,
+        ${escapeId('content')} TEXT,
+        PRIMARY KEY(${escapeId('snapshotKey')})
+        )`
+    )
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-lite/src/init.ts:62]: Dependent calls.
If the first call is successful, and the second is unsuccessful, then it is difficult to repeat the creation

File: packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts [hunk 3]
```diff
@@ -1,28 +1,56 @@
...
+  const secretsTableNameAsId: string = escapeId(secretsTableName)
 
-  log.debug(`secrets store database tables are dropped`)
-}
+  const statements: string[] = [
+    `DROP TABLE IF EXISTS ${freezeTableNameAsId}`,
+    `DROP TABLE ${threadsTableNameAsId}`,
+    `DROP TABLE ${eventsTableNameAsId}`,
+    `DROP TABLE ${snapshotsTableNameAsId}`,
+    `DROP TABLE IF EXISTS ${secretsTableNameAsId}`,
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts:28]: Inconsistently. "DROP TABLE" <-> "DROP TABLE IF EXISTS"

File: packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts [hunk 4]
```diff
@@ -1,28 +1,56 @@
...
+        }
+        errors.push(error)
+      }
+    }
+  }
 
-  log.debug(`dropping the event store`)
-  await Promise.all([dropEventStore(pool), dropSecretsStore(pool)])
-  log.debug(`the event store dropped`)
+  if (errors.length > 0) {
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts:51]: Revert logs

File: packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts [hunk 5]
```diff
@@ -1,28 +1,56 @@
...
-import dropEventStore from './js/drop'
+import { EventstoreResourceNotExistError } from 'resolve-eventstore-base'
+import { EOL } from 'os'
+import getLog from './get-log'
 import { AdapterPool } from './types'
 
-const dropSecretsStore = async (pool: AdapterPool): Promise<any> => {
-  const log = getLog('dropSecretsStore')
-
-  log.debug(`dropping secrets store database tables`)
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts:8]: Revert logs

File: packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts [hunk 6]
```diff
@@ -1,28 +1,56 @@
...
-  log.verbose(`secretsTableName: ${tableName}`)
 
-  await connection.execute(`DROP TABLE IF EXISTS ${escapeId(tableName)}`)
+  const eventsTableNameAsId: string = escapeId(eventsTableName)
+  const freezeTableNameAsId: string = escapeId(`${eventsTableName}-freeze`)
+  const threadsTableNameAsId: string = escapeId(`${eventsTableName}-threads`)
+  const snapshotsTableNameAsId: string = escapeId(snapshotsTableName)
+  const secretsTableNameAsId: string = escapeId(secretsTableName)
 
-  log.debug(`secrets store database tables are dropped`)
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts:17]: Revert logs

File: packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts [hunk 7]
```diff
@@ -1,28 +1,56 @@
...
+          log.verbose(error.stack)
+        }
+        errors.push(error)
+      }
+    }
+  }
 
-  log.debug(`dropping the event store`)
-  await Promise.all([dropEventStore(pool), dropSecretsStore(pool)])
-  log.debug(`the event store dropped`)
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-mysql/src/drop.ts:25]: Revert logs

File: packages/adapters/eventstore-adapters/resolve-eventstore-postgresql-serverless/src/drop.ts [hunk 8]
```diff
@@ -55,31 +50,60 @@ const dropSecretsStore = async (pool: AdapterPool): Promise<any> => {
...
+    `DROP INDEX IF EXISTS ${databaseNameAsId}.${aggregateIndexName}`,
+    `DROP INDEX IF EXISTS ${databaseNameAsId}.${aggregateVersionIndexName}`,
+    `DROP INDEX IF EXISTS ${databaseNameAsId}.${typeIndexName}`,
+    `DROP INDEX IF EXISTS ${databaseNameAsId}.${timestampIndexName}`,
+
+    `DROP TABLE ${databaseNameAsId}.${threadsTableNameAsId}`,
+
+    `DROP TABLE ${databaseNameAsId}.${snapshotsTableNameAsId}`,
+
+    `DROP TABLE IF EXISTS ${databaseNameAsId}.${freezeTableNameAsId}`,
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-postgresql-serverless/src/drop.ts:82]: Inconsistently

File: packages/adapters/eventstore-adapters/resolve-eventstore-postgresql-serverless/src/save-snapshot.ts [hunk 9]
```diff
@@ -38,8 +43,7 @@ const saveSnapshot = async (pool, snapshotKey, content) =>
             await executeStatement(
               `UPDATE ${databaseNameAsId}.${snapshotsTableNameAsId}
             SET "snapshotContent" = "snapshotContent" || ${escape(chunk)}
-            WHERE "snapshotKey" = ${escape(snapshotKey)}`,
-              transactionId
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-postgresql-serverless/src/save-snapshot.ts:42]: Where "transactionId"?

File: packages/adapters/eventstore-adapters/resolve-eventstore-postgresql-serverless/src/save-snapshot.ts [hunk 10]
```diff
@@ -49,8 +53,7 @@ const saveSnapshot = async (pool, snapshotKey, content) =>
             )
             VALUES(${escape(snapshotKey)}, ${escape(chunk)})
             ON CONFLICT ("snapshotKey") DO UPDATE
-            SET "snapshotContent" = ${escape(chunk)}`,
-              transactionId
+            SET "snapshotContent" = ${escape(chunk)}`
```
Comment (MrCheater) [inline packages/adapters/eventstore-adapters/resolve-eventstore-postgresql-serverless/src/save-snapshot.ts:56]: Where "transactionId"?

File: packages/adapters/eventstore-adapters/resolve-eventstore-mysql/test/__snapshots__/drop.unit.test.ts.snap [hunk 11]
```diff
@@ -1,9 +1,3 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`secrets store dropped: drop table with keys 1`] = `
-Array [
-  Array [
-    "DROP TABLE IF EXISTS \\"secrets-database-escaped-id\\"",
-  ],
-]
-`;
+exports[`secrets store dropped: drop table with keys 1`] = `Array []`;
```
Comment (timbset) [inline packages/adapters/eventstore-adapters/resolve-eventstore-mysql/test/__snapshots__/drop.unit.test.ts.snap:3]: Behavior should not have changed if it's just a refactoring

**Review by timbset — APPROVED** [review]

**Review by MrCheater — APPROVED** [review]

**Pull request merged**